ui/lib/kv/addon/routes/list-directory.js

```diff
@@ -14,25 +14,40 @@ export default class KvSecretsListRoute extends Route {
   @service router;
   @service secretMountPath;
 
-  getPathToSecretFromUrl() {
-    const { path_to_secret } = this.paramsFor('list-directory');
-    return path_to_secret ? normalizePath(path_to_secret) : '';
-  }
+  queryParams = {
+    // TODO kv engine cleanup: remove by filtering in the component
```

Comment:
Not sure we can do this, especially if we are going to be using `lazyPaginatedQuery` (which we likely will need to do -- KV secret lists can be quite long IIRC)

Reply:
Yeah, I kind of wondered. I played around with this and couldn't get it to work without pagination so I can imagine it's only going to get harder. I'll remove the TODO. I've got some notes about addressing this in a nice-to-have jira ticket.